src/fqm.ts

```diff
@@ -42,39 +42,54 @@ export const calculateMeasureReports = async (
 };
 
 /**
- * Populates the _type parameter used in a bulk data export request. Retrieves the data
- * requirements for the given measure bundle using the fqm-execution API function. Extracts
- * the resource types from the data requirements. Throws error if data requirements are not
- * defined on the results of the API function.
+ * Populates the _type/_typeFilter parameters used in a bulk data export request. Retrieves the data
+ * requirements for the given measure bundle using the fqm-execution API function to use for
+ * _type and _typeFilter query construction. Throws error if data requirements are not defined on the results
+ * of the API function.
  * @param measureBundle FHIR measure bundle
+ * @param autoType boolean specifying whether we want to auto-generate _type
+ * @param autoTypeFilter boolean specifying whether we want to auto-generate _typeFilter
  * @param options fqm-execution calculation options
  */
-export const retrieveTypeFromMeasureBundle = async (
+export const retrieveParamsFromMeasureBundle = async (
   measureBundle: fhir4.Bundle,
+  autoType: boolean,
+  autoTypeFilter: boolean,
   options: CalculatorTypes.CalculationOptions = {}
 ) => {
   const dr = await Calculator.calculateDataRequirements(measureBundle, options);
   if (!dr.results.dataRequirement) {
     throw new Error('Data requirements array is not defined for the Library. Aborting $export request.');
   }
-  return constructTypeQueryFromRequirements(dr.results.dataRequirement);
+  return constructParamsFromRequirements(dr.results.dataRequirement, autoType, autoTypeFilter);
 };
 
 /**
- * Constructs the _type parameter used in a bulk data export request by parsing the
- * data requirements for the measure bundle.
+ * Constructs the _type and _typeFilter parameters used in a bulk data export request by parsing the
+ * data requirements from the measure bundle.
  * @param dataRequirements data requirements retrieved from measure bundle
+ * @param autoType boolean specifying whether we want to auto-generate _type
+ * @param autoTypeFilter boolean specifying whether we want to auto-generate _typeFilter
  */
-export const constructTypeQueryFromRequirements = (dataRequirements: fhir4.DataRequirement[]) => {
+export const constructParamsFromRequirements = (
+  dataRequirements: fhir4.DataRequirement[],
+  autoType: boolean,
+  autoTypeFilter: boolean
+) => {
+  const queries: { type: string; params: Record<string, string> }[] = [];
   const types: string[] = [];
-
   dataRequirements.forEach((dr) => {
     if (dr.type) {
       types.push(dr.type);
+      const query: { type: string; params: Record<string, string> } = { type: dr.type, params: {} };
+      if (dr?.codeFilter?.[0]?.valueSet) {
```

Comment:
This seems to limited the _typeFilter to only include valueset filters that happen to be in the 0 index for the Data Requirement.  This does not support Data Requirements for elements that have Direct Reference Codes.  For example, this Observation for "Functional Assessment of Chronic Illness Therapy - Palliative Care Questionnaire (FACIT-Pal)" which uses the Direct Reference Code '71007-9'.

![Screenshot 2023-08-17 at 12 48 31 PM](https://github.com/bulk-dqm/bulk-data-export-client/assets/8173551/7b08ba6c-6f62-44d1-b1c4-2faaf96632f1)

This will also omit any code filters that are not in the index of 0.  For example, this same Observation has a status element which "could" also be included in the filter.  I do feel like its ok to omit these coded elements from the automated filter.

![Screenshot 2023-08-17 at 12 52 29 PM](https://github.com/bulk-dqm/bulk-data-export-client/assets/8173551/d7a8ad56-2661-42f6-b9e8-60a2edf05ef5)

Reply:
I updated the logic to loop through the `codeFilter`s rather than omitting any code filter that is not at index 0. I also added support for direct reference codes, although the Abacus export server does not support filtering via direct reference codes. To check the logic here, I suggest looking at the comma-delimited `_typeFilter` string and checking that we are capturing the correct information, since we cannot check what data gets returned from the filtering at this time. I will make a note to add support for direct reference codes on the server.